packages/rln/src/credentials_manager.ts

```diff
@@ -0,0 +1,251 @@
+
+    if (chainId && chainId !== currentChainId) {
+      log.error(
+        `Chain ID mismatch: contract=${chainId}, current=${currentChainId}`
+      );
+      throw Error(
+        `Failed to start RLN contract, chain ID of contract is different from current one: contract-${chainId}, current network-${currentChainId}`
+      );
+    }
+
+    return {
+      signer,
+      address
+    };
+  }
+
+  private static async decryptCredentialsIfNeeded(
+    credentials?: EncryptedCredentials | DecryptedCredentials
+  ): Promise<{ credentials?: DecryptedCredentials; keystore?: Keystore }> {
+    if (!credentials) {
+      log.info("No credentials provided");
+      return {};
+    }
+
+    if ("identity" in credentials) {
+      log.info("Using already decrypted credentials");
+      return { credentials };
+    }
+
+    log.info("Attempting to decrypt credentials");
+    const keystore = Keystore.fromString(credentials.keystore);
+
+    if (!keystore) {
+      log.warn("Failed to create keystore from string");
+      return {};
+    }
+
+    try {
+      const decryptedCredentials = await keystore.readCredential(
+        credentials.id,
+        credentials.password
+      );
+      log.info(`Successfully decrypted credentials with ID: ${credentials.id}`);
+
+      return {
+        keystore,
+        credentials: decryptedCredentials
+      };
+    } catch (error) {
+      log.error("Failed to decrypt credentials", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Generates an identity credential from a seed string
+   * This is a pure implementation that doesn't rely on Zerokit
+   * @param seed A string seed to generate the identity from
+   * @returns IdentityCredential
+   */
+  private generateSeededIdentityCredential(seed: string): IdentityCredential {
+    log.info("Generating seeded identity credential");
+    // Convert the seed to bytes
+    const encoder = new TextEncoder();
+    const seedBytes = encoder.encode(seed);
+
+    // Generate deterministic values using HMAC-SHA256
+    // We use different context strings for each component to ensure they're different
+    const idTrapdoor = hmac(sha256, seedBytes, encoder.encode("IDTrapdoor"));
+    const idNullifier = hmac(sha256, seedBytes, encoder.encode("IDNullifier"));
+
+    // Generate IDSecretHash as a hash of IDTrapdoor and IDNullifier
+    const combinedBytes = new Uint8Array([...idTrapdoor, ...idNullifier]);
+    const idSecretHash = sha256(combinedBytes);
+
+    // Generate IDCommitment as a hash of IDSecretHash
+    const idCommitment = sha256(idSecretHash);
+
+    // Convert IDCommitment to BigInt
+    const idCommitmentBigInt = buildBigIntFromUint8Array(idCommitment);
+
+    log.info("Successfully generated identity credential");
+    return new IdentityCredential(
+      idTrapdoor,
+      idNullifier,
+      idSecretHash,
+      idCommitment,
+      idCommitmentBigInt
+    );
+  }
+
+  public async registerMembership(
```

Comment:
move `public` methods above `private` 
I think we didn't enforce this by `eslint` yet, but I remember we agreed to it within the team

https://github.com/waku-org/js-waku/issues/2334

Reply:
https://github.com/waku-org/js-waku/pull/2295/commits/6997987a60a52f1382230ff7b64b25606610534f